src/main/java/org/stellar/sdk/SorobanServer.java

```diff
@@ -0,0 +1,344 @@
+import org.stellar.sdk.responses.sorobanrpc.SendTransactionResponse;
+import org.stellar.sdk.responses.sorobanrpc.SimulateTransactionResponse;
+import org.stellar.sdk.responses.sorobanrpc.SorobanRpcResponse;
+import org.stellar.sdk.xdr.ContractDataDurability;
+import org.stellar.sdk.xdr.ContractEntryBodyType;
+import org.stellar.sdk.xdr.LedgerEntry;
+import org.stellar.sdk.xdr.LedgerEntryType;
+import org.stellar.sdk.xdr.LedgerKey;
+import org.stellar.sdk.xdr.SCVal;
+import org.stellar.sdk.xdr.SorobanAuthorizationEntry;
+import org.stellar.sdk.xdr.XdrDataInputStream;
+import org.stellar.sdk.xdr.XdrDataOutputStream;
+
+/**
+ * Main class used to connect to the Soroban-RPC instance and exposes an interface for requests to
+ * that instance.
+ */
+public class SorobanServer implements Closeable {
+  private static final int SUBMIT_TRANSACTION_TIMEOUT = 60; // seconds
+  private static final int CONNECT_TIMEOUT = 10; // seconds
+  private final HttpUrl serverURI;
+  private final OkHttpClient httpClient;
+  private final Gson gson = new Gson();
+
+  /**
+   * Creates a new SorobanServer instance.
+   *
+   * @param serverURI The URI of the Soroban-RPC instance to connect to.
+   */
+  public SorobanServer(String serverURI) {
+    this(
+        serverURI,
+        new OkHttpClient.Builder()
+            .addInterceptor(new ClientIdentificationInterceptor())
+            .connectTimeout(CONNECT_TIMEOUT, TimeUnit.SECONDS)
+            .readTimeout(SUBMIT_TRANSACTION_TIMEOUT, TimeUnit.SECONDS)
+            .retryOnConnectionFailure(true)
+            .build());
+  }
+
+  public SorobanServer(String serverURI, OkHttpClient httpClient) {
+    this.serverURI = HttpUrl.parse(serverURI);
+    this.httpClient = httpClient;
+  }
+
+  public TransactionBuilderAccount getAccount(String accountId)
+      throws IOException, LedgerEntryNotFoundException {
+    LedgerKey.LedgerKeyAccount ledgerKeyAccount =
+        new LedgerKey.LedgerKeyAccount.Builder()
+            .accountID(KeyPair.fromAccountId(accountId).getXdrAccountId())
+            .build();
+    LedgerKey ledgerKey =
+        new LedgerKey.Builder()
+            .account(ledgerKeyAccount)
+            .discriminant(LedgerEntryType.ACCOUNT)
+            .build();
+    GetLedgerEntriesResponse getLedgerEntriesResponse =
+        this.getLedgerEntries(Collections.singleton(ledgerKey));
+    List<GetLedgerEntriesResponse.LedgerEntryResult> entries =
+        getLedgerEntriesResponse.getEntries();
+    if (entries == null || entries.isEmpty()) {
+      throw new LedgerEntryNotFoundException(ledgerKeyToXdrBase64(ledgerKey));
+    }
+    LedgerEntry.LedgerEntryData ledgerEntryData =
+        ledgerEntryDataFromXdrBase64(entries.get(0).getXdr());
+    long sequence = ledgerEntryData.getAccount().getSeqNum().getSequenceNumber().getInt64();
+    return new Account(accountId, sequence);
+  }
+
+  public GetHealthResponse getHealth() throws IOException {
+    return this.<Void, GetHealthResponse>sendRequest(
```

Comment:
can remove the `this` references, that scope is implied when any non-static method invokes another non-static method of class.

Reply:
Yes, but I think adding "this" is also very common. If you insist on removing it, I can do that.

Reply:
it's not common to use `this` when the variable scope doesn't require it, one example of usage is when a method level variable shadows a class variable, but the usage here in this scope is not required and results in some code bloat.